proj/src/main/webapp/script.js

```diff
@@ -69,4 +72,35 @@ function pushChatMessage() {
     const message = document.getElementById('message-input').value;
     // push message to datastore
     messageRef.push(message);
-} 
+}
+
+var messages;
+function addMoreMessagesAtTheTop() {
+    const dbRefObject = firebase.database().ref(path);
+    const chat = document.getElementById('chat-as-list');
+    if (chat.scrollTop === 0) {
+        const oldScrollHeight = chat.scrollHeight;
+        console.log(oldScrollHeight);
+        // because we don't add the last child, add one to the limit
+        dbRefObject.orderByKey().endAt(FIRST_CHILD_KEY).limitToLast(LIMIT + 1).once('value', snap => {
+            FIRST_CHILD_KEY = null;
+            addMessagesToListElement(snap.val(), chat.firstChild, oldScrollHeight);
+        });
+    }
+}
+
+function addMessagesToListElement(messages, firstChild, oldScrollHeight) {
+    const chat = document.getElementById('chat-as-list');
+    for (var key in messages) {
+        if (messages.hasOwnProperty(key)) {
+            if (!FIRST_CHILD_KEY) {
+                FIRST_CHILD_KEY = key;
+                continue;
```

Comment:
This can be an if-else rather than using continue.

Reply:
fixed!